### PR TITLE
Add reqmgr2ms-pileup service

### LINKED ISSUE #11423: [MS] Create a new microservice skeleton called MSPileup
**Impact of the new feature**
 New WMCore microservice

**Is your feature request related to a problem? Please describe.**
Creation of a new microservice to make pileup management more efficient and easier.
Subtask of https://github.com/dmwm/WMCore/issues/9779

**Describe the solution you'd like**
As part of a better pileup data management within the WM system, we need to create the foundations of a new microservice called MSPileup (or any other more meaningful name). This ticket does not require any service logic to be implemented at this moment, but we should provide:
* a new subpackage/module under WMCore/Microservices
* deployment scripts under the deployment and services_config repository
* dockerfile and yaml manifest under the CMSKubernetes
* create frontend/nginx rules
* definition of this new service under the setup_dependencies.py
* automatic build and upload of this new service (via GH actions)
* REST APIs for `status` and `info` should be functional (same behavior as for the other microservices)

**Describe alternatives you've considered**
None

**Additional context**
None

### PR DESCRIPTION
**Pull request by vkuznet**
Fixes #11423 

#### Status
in development

#### Description
First draft for MSPileup service

#### Is it backward compatible (if not, which system it affects?)
YES

#### Related PRs
- [Dockerfile](https://github.com/dmwm/CMSKubernetes/blob/master/docker/pypi/reqmgr2ms-pileup/Dockerfile)
- [apache frontend rules](https://github.com/dmwm/deployment/pull/1230)
- reqmgr2ms-pileup configuration draft can be found in this [config.py](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/blob/test/reqmgr2ms-pileup/config.py)
- [reqmgr2ms-pileup.yaml](https://github.com/dmwm/CMSKubernetes/blob/master/kubernetes/cmsweb/services/reqmgr2ms-pileup.yaml) k8s manifest file to deploy service to k8s cluster
- reqmgr2ms-pileup configuration draft can be found in gitlab [services_configs/reqmgr2ms-pileup/config.py](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/blob/test/reqmgr2ms-pileup/config.py) repo
I used port 8249 as new port number for this service

The service skeleton is also tested with local setup using curl. see https://github.com/dmwm/WMCore/pull/11430#issuecomment-1376324420

Additional changes made by @arooshap :
- For the cmsweb cluster: [services_config/frontend-ds](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/commit/ce8c18838e7dcd22162aaf61834ea48c64a48dd7).
- For the preprod cluster: [services_config/frontend-ds](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/commit/fa4cd17cd67e8d26b5b46916fab145e4171ca288).
- and for the test cluster: [services_config/frontend-ds](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/commit/bd07af9c422fc2dd3cd45428d7fec7eb1ca3badf).

### CODE REVIEW

**Comment by vkuznet** [comment]
@amaltaro , @todor-ivanov I made initial attempt to create MSPileup service. The changes are spread across multiple projects and I'll summarize them here:
- initial code base for MSPileup service, this PR
- changes to setup_dependencies.py, this PR
- changes to CI/CD, this PR
- [Dockerfile](https://github.com/dmwm/CMSKubernetes/blob/master/docker/pypi/reqmgr2ms-pileup/Dockerfile)
- [frontend redirect rules](https://github.com/dmwm/deployment/pull/1229) in deployment repo
- reqmgr2ms-pileup configuration draft can be found in gitlab [services_configs/reqmgr2ms-pileup/config.py](https://gitlab.cern.ch/cmsweb-k8s/services_config/-/blob/test/reqmgr2ms-pileup/config.py) repo
  - I used port 8249 as new port number for this service
- [reqmgr2ms-pileup.yaml](https://github.com/dmwm/CMSKubernetes/blob/master/kubernetes/cmsweb/services/reqmgr2ms-pileup.yaml) k8s manifest file to deploy service to k8s cluster

I saw that pycurl_manager failed during unit tests which seems completely irrelevant here and rather an issue with access to cms-gwmsmon.cern.ch service, see [here](https://cmssdt.cern.ch/dmwm-jenkins/job/DMWM-WMCore-PR-test/13865/testReport/junit/WMCore_t.Services_t.pycurl_manager_t/PyCurlManager/testSingle/). I suggest that someone decide if we need to use this service in pycurl tests anymore but I doubt it has anything to do with this PR.

Said that, I need your initial review to proceed. If anything is missing or should be added feel free to point me to it. Meanwhile I tested new service in local setup and was able to get proper values from`/ms-pileup/status` and `/ms-pileup/info` end-points:
```
curl -v http://localhost:8249/ms-pileup/data/status
...
< HTTP/1.1 200 OK
...
{"result": [
 {
  "wmcore_version": "2.1.6rc5",
  "microservice_version": "2.1.6rc5",
  "microservice": "MSManager",
  "status": "OK"
}]}

curl -v http://localhost:8249/ms-pileup/data/info
...
< HTTP/1.1 200 OK
...
{"result": [
 {
  "wmcore_version": "2.1.6rc5",
  "microservice_version": "2.1.6rc5",
  "microservice": "MSManager",
  "request": null,
  "transferDoc": null
}]}
```

File: .github/workflows/pypi_build_and_images.yaml [hunk 1]
```diff
@@ -27,7 +27,7 @@ jobs:
     strategy:
       matrix:
         target: [wmagent, wmagent-devtools, wmcore, reqmon, reqmgr2, global-workqueue, acdcserver, reqmgr2ms-unmerged,
-                 reqmgr2ms-output, reqmgr2ms-rulecleaner, reqmgr2ms-transferor, reqmgr2ms-monitor]
+                 reqmgr2ms-output reqmgr2ms-pileup, reqmgr2ms-rulecleaner, reqmgr2ms-transferor, reqmgr2ms-monitor]
```
Comment (amaltaro) [inline .github/workflows/pypi_build_and_images.yaml:30]: Missing comma here.

**Review by amaltaro — CHANGES_REQUESTED** [review]
@vkuznet all these changes are looking pretty good. I made a comment along the code that needs fixing.

In addition to that:
* on the Dockerfile, you have a typo in there `RUN pip install reqmgr2ms-pielup==$TAG`
* for the config.py, in services_config, I would suggest to start it with the bare minimum configuration at the moment and add attributes as we need them. I can get back to you with a candidate configuration soon.

Can you please also make sure that your comment in the middle of this PR gets incorporated to the very first comment in the PR?

File: .github/workflows/pypi_build_and_images.yaml (hunk 1 above)
Comment (vkuznet) [reply .github/workflows/pypi_build_and_images.yaml]: done

**Comment by vkuznet** [comment]
Alan, thanks for spotting missing comma and typo. I fixed both.

**Comment by vkuznet** [comment]
Alan, I squashed the changes, updated PR description, and I think this PR is ready to go. Feel free to review again and merge. The confit file can be updated separately via gitlab PR and you may submit your version over there. Once this PR is merged I can continue on separate PRs for other (implementation) parts of new service.

**Review by amaltaro — APPROVED** [review]
It looks all good, thanks!

**Comment by amaltaro** [comment]
PS.: I am removing this PR from the project because the project is supposed to be used only for GH issues. PR labels usually start with "PR:" as well, so I am removing those that are unneeded.

**Pull request merged**